src/components/tasks/TaskSearch/TaskSearch.tsx

```diff
@@ -16,84 +23,316 @@ const initialSuggestionCoordinates: SuggestionCoordinates = {
 };
 
 type TaskSearchProps = {
-    onSelect: (tab: Tab) => void;
+    onFilterDropdownSelect: (tab: Tab) => void;
+    filterDropdownActiveTab?: Tab;
     inputValue: string;
-    activeTab?: Tab;
-    onInputChange: (value: string) => void;
     onClickSearchButton: (param?: string) => void;
 };
 
 const TaskSearch = ({
-    onSelect,
+    onFilterDropdownSelect,
+    filterDropdownActiveTab,
     inputValue,
-    activeTab,
-    onInputChange,
     onClickSearchButton,
 }: TaskSearchProps) => {
-    const [modalOpen, setModalOpen] = useState(false);
+    const [filterDropdownModelOpen, setFilterDropdownModelOpen] =
+        useState(false);
+
+    const userInputRef = useRef<HTMLInputElement>(null);
+    const [typedInput, setTypedInput] = useState('');
+    const defferedUserInput: string = useDebounce(typedInput, 300);
+
+    const [selectedFilters, setSelectedFilters] = useState<
+        Array<TaskSearchOption>
+    >(convertStringToOptions(inputValue));
+    const [onEditSelectedFilterValue, setOnEditSelectedFilterValue] =
+        useState<string>('');
+    const [onEditSelectedFilterIndex, setOnEditSelectedFilterIndex] =
+        useState<false | number>(false);
+    const [onRemoveSelectedFilterIndex, setOnRemoveSelectedFilterIndex] =
+        useState(-1);
+    const defferedPillValue = useDebounce(onEditSelectedFilterValue, 300);
+
+    const [
+        activeFilterSuggestionDropdownIndex,
+        setActiveFilterSuggestionDropdownIndex,
+    ] = useState(-1);
```

Comment:
state variables like `onEditSelectedFilterIndex`, `onRemoveSelectedFilterIndex`, and `activeFilterSuggestionDropdownIndex` are a bit confusing. Consider using more descriptive names.